linux/net/rina/ipcp-instances.h

```diff
@@ -94,9 +94,19 @@ struct efcp_config {
         /* The data transfer constants */
         struct dt_cons * dt_cons;
 
+        /* Left here for phase 2 */
         struct policy * unknown_flow;
 };
 
+/* Represents the configuration of the RMT */
+struct rmt_config {
+	/* The PS name for the PDU Forwarding Table (Next hop) */
+	struct policy * pft_policy_set;
+
+	/* The PS name for the RMT */
+	struct policy * rmt_policy_set;
+};
+
 /* Represents a DIF configuration (policies, parameters, etc) */
```

Comment:
Yes, the question arises from the fact that I know how SDK (policy-sets and
plugins) is implemented :)
I get what this is, but the question remains the same: what's the point in
storing those strings there? We already have policy-set factories (with
names and factory methods) stored in the per-component policy-set lists. It
seems unused and redundant information.

2015-06-03 12:16 GMT+02:00 Eduard Grasa <notifications@github.com>:

> In linux/net/rina/ipcp-instances.h
> <https://github.com/IRATI/stack/pull/620#discussion_r31609938>:
>
> >          struct policy * unknown_flow;
> >  };
> >
> > +/* Represents the configuration of the RMT */
> > +struct rmt_config {
> > +	/* The PS name for the PDU Forwarding Table (Next hop) */
> > +	struct policy * pft_policy_set;
> > +
> > +	/* The PS name for the RMT */
> > +	struct policy * rmt_policy_set;
> > +};
> > +
> >  /* Represents a DIF configuration (policies, parameters, etc) */
>
> This is the DIF configuration that comes from user-space, and the policies
> the RMT has to be configured with (when assign-to-dif is called). You can
> have a number of RMT policy-sets published by plugins in the kernel (for
> different DIFs, for example), this is the configuration of an RMT for a
> particular DIF. 2015-06-03 11:58 GMT+02:00 Vincenzo Maffione <
> notifications@github.com>:
> … <#14db8ec745f67bc0_>
>  In linux/net/rina/ipcp-instances.h <
> https://github.com/IRATI/stack/pull/620#discussion_r31608763>: > struct
> policy * unknown_flow; > }; > > +/* Represents the configuration of the RMT
> */ > +struct rmt_config { > + /* The PS name for the PDU Forwarding Table
> (Next hop) */ > + struct policy * pft_policy_set; > + > + /* The PS name
> for the RMT */ > + struct policy * rmt_policy_set; > +}; > + > /*
> Represents a DIF configuration (policies, parameters, etc) */ What's the
> point of storing the policy names there? Remember that policy-set names are
> already stored in the per-component "available policy set list", as soon as
> they are published by plugins. — Reply to this email directly or view it on
> GitHub <https://github.com/IRATI/stack/pull/620/files#r31608763>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/IRATI/stack/pull/620/files#r31609938>.
>



-- 
Vincenzo Maffione